semeio/jobs/ots/ots.py

```diff
@@ -0,0 +1,597 @@
+                        dt.now(), vd
+                    )
+                )
+                sys.stdout.flush()
+
+            self.add_survey(vn, vd)
+
+            num_points_calculated = 0
+            for point in range(num_points):
+
+                if not np.isnan(surface.z[point]):
+                    num_points_calculated += 1
+                    r1 = (surface.x[point], surface.y[point], 0)
+                    r2 = (
+                        surface.x[point],
+                        surface.y[point],
+                        surface.z[point] - self._seabed,
+                    )
+                    # subsidence and displacement have opposite sign
+                    # should have minus on dz1 and dz2 here, more efficient
+                    # when calculating ts
+                    dz1 = self.subsidence.evalGeertsma(
+                        base_survey=vn,
+                        monitor_survey=None,
+                        pos=r1,
+                        youngs_modulus=self._youngs_modulus,
+                        poisson_ratio=self._poisson_ratio,
+                        seabed=self._seabed,
+                    )
+                    dz2 = self.subsidence.evalGeertsma(
+                        base_survey=vn,
+                        monitor_survey=None,
+                        pos=r2,
+                        youngs_modulus=self._youngs_modulus,
+                        poisson_ratio=self._poisson_ratio,
+                        seabed=self._seabed,
+                    )
+                    du[iv, point] = dz2 - dz1
+
+        for ivp, vp in enumerate(vintage_pairs):
+            if self._verbose:
+                if self._convention == 1:
+                    print(
+                        "{:%x %X} TS_SIMPLE: Calculating shift"
+                        " {:%Y.%m.%d}-{:%Y.%m.%d} in {} points".format(
+                            dt.now(), vp[0], vp[1], num_points_calculated
+                        )
+                    )
+                if self._convention == -1:
+                    print(
+                        "{:%x %X} TS_SIMPLE: Calculating shift"
+                        " {:%Y.%m.%d}-{:%Y.%m.%d} in {} points".format(
+                            dt.now(), vp[1], vp[0], num_points_calculated
+                        )
+                    )
+                if num_points_calculated == 0:
+                    print("Shift was calculated in 0 points.", file=sys.stderr)
+                    print("Consider changing --apply_mapaxes value.")
+
+                sys.stdout.flush()
+            base_index = vintages.date.index(vp[0])
+            monitor_index = vintages.date.index(vp[1])
+            ts = -self._r_factor * (du[monitor_index] - du[base_index])
+            self._divide_negative_shift(ts)
+
+            ts = ts * self._convention
+
+            ts_surfaces.append(self._create_surface(ts))
+
+        return ts_surfaces
+
+    def geertsma_ts(self, vintage_pairs):
```

Comment:
So this is the function that does the proper time shift evaluation, while the other two are turning some corners. I am not sure if it is as easy to merge them all three into a function, but all three functions are doing the same kind of logging. Splitting the content of the functions a bit will certainly help on that.

Reply:
I see your point, but prefer to have it split. Especially if logging message is slightly different for each of the four methods. Splitting them would bring some additional ifs and more parameters for the functions. What can be done is to group logging code into semantically same-y methods.

Reply:
Something like:

```
    def _report(self, func_name, base, monitor, num_points_calculated):
        if self._convention == 1:
            start_date, end_date = monitor, base
        elif self._convention == -1:
            start_date, end_date = base, monitor
        else:
            raise ValueError("Convention must be 1 or -1, was {}".format(self._convention))
        logging.DEBUG(
                "{:%x %X} TS: Calculating shift"
                " {:%Y.%m.%d}-{:%Y.%m.%d} in {} points".format(
                    dt.now(), start_date, end_date, num_points_calculated
                )
            )
```
And have that called in each of the four places.

Reply:
Agree, worked in already!